sdk/keyvault/Azure.Security.KeyVault.Administration/src/KeyVaultBackupClient.cs

```diff
@@ -335,6 +335,68 @@ internal virtual Response<RestoreDetailsInternal> GetRestoreDetails(string jobId
             }
         }
 
+        /// <summary>
+        /// Returns the details of selective restore operation.
+        /// </summary>
+        /// <param name="jobId"> The Job Id returned part of the full restore operation. </param>
+        /// <param name="cancellationToken"> The cancellation token to use. </param>
+        /// <exception cref="ArgumentNullException"><paramref name="jobId"/> is null.</exception>
+        /// <exception cref="RequestFailedException">The server returned an error. See <see cref="Exception.Message"/> for details returned from the server.</exception>
+        internal virtual async Task<Response<SelectiveKeyRestoreDetailsInternal>> GetSelectiveKeyRestoreDetailsAsync(string jobId, CancellationToken cancellationToken = default)
+        {
+            using DiagnosticScope scope = _diagnostics.CreateScope($"{nameof(KeyVaultBackupClient)}.{nameof(GetRestoreDetails)}");
+            scope.Start();
+            try
+            {
+                var restoreResult = await _restClient.RestoreStatusAsync(VaultUri.AbsoluteUri, jobId, cancellationToken).ConfigureAwait(false);
+                return Response.FromValue(
+                    new SelectiveKeyRestoreDetailsInternal(
```

Comment:
Nit: rather than duplicating this, maybe declare a static factory method or constructor overload to just take in the `restoreResult` type and consolidate the logic to the class itself.